pulsar-websocket/src/main/java/org/apache/pulsar/websocket/ProducerHandler.java

```diff
@@ -298,7 +298,7 @@ private void updateSentMsgStats(long msgSize, long latencyUsec) {
         }
 
         if (queryParams.containsKey("initialSequenceId")) {
-            builder.initialSequenceId(Long.parseLong("initialSequenceId"));
+            builder.initialSequenceId(Long.parseLong(queryParams.get("initialSequenceId")));
```

Comment:
Add a test for `getProducerBuilder` methods? We can verify the producer applied all we gave configurations.

Reply:
The `getProducerBuilder` cannot be tested simply. The parameter validation happens in `ProducerBuilder#create`. However the `PulsarClient`'s service url cannot be mocked because it retrieves the url related params from ZK, see `WebSocketService#getPulsarClient` for details.

Therefore, it needs an integration test which may be beyond this PR's scope of work. Also I found there's no WebSocket related integration tests. If you think it's proper to add an WebSocket integration test in this PR, I'll take some time to do it.

Reply:
We don't need to validate the params. We can simply mock the queryParams which is a hashmap and put the arguments into the map. Then when we can get the values in the `getProducerBuilder`. We can make sure the ProducerBuilder contains the configurations which we pass to the hashmap.

Currently, the `getProducerBuilder` can not get the right builder with the `queryParams`, so we can make sure it can get the right builder with the given params.

Reply:
I mean something like this:
```
    @Test
    public void testGetTheRightBuilder() {
        HttpServletRequest request = mock(HttpServletRequest.class);
        when(request.getParameterMap()).thenReturn(mockedmap)
        ProducerHandler producerHandler = new ProducerHandler(null, request, null);
        ProducerBuilderImpl<byte[]> builder = (ProducerBuilderImpl<byte[]>) producerHandler.getProducerBuilder(null);
        builder.getConf().getProducerName();
    }
```

Reply:
But `ProducerBuilder` just provides getters but not setters. So we need to call `create` and it will fail if `PulsarClient`'s service url is not correct.

Reply:
In addition, `getProducerBuilder` is private. Even if change it to `public`, it must be called after `ProducerHandler` is constructed. However, the constructor calls `getProducerBuilder` and `ProducerBuilder#create`. This behavior cannot be changed even with a mocked `WebSocketService` and `HttpServletRequest`.

Reply:
I am not sure can we cast `ProduceBuilder` to `ProducerBuilderImpl` to get some properties? 
```
        ProducerBuilderImpl<byte[]> builder = (ProducerBuilderImpl<byte[]>) producerHandler.getProducerBuilder(null);
        builder.getConf().getProducerName();
```

Reply:
Since the exception is thrown by `getProducerBuilder`, I think I can add a test to check if any `NumberFormatException` or `IllegalArgumentException`.

Reply:
You can use`mock` to get every behavior you want.

Reply:
If you need to use the `getProducerBuilder` method, I think it's ok to make it to package private not public.

Reply:
Ok, I'll add a test soon.